Source/Core/UICommon/CommandLineParse.cpp

```diff
@@ -34,6 +38,52 @@ class CommandLineConfigLayerLoader final : public Config::ConfigLayerLoader
                             ValueToString(audio_backend == "HLE"));
     }
 
+    if (netplay_host.has_value())
+    {
+      const std::string traversal_choice = Config::Get(Config::NETPLAY_TRAVERSAL_CHOICE);
+      const bool is_traversal = traversal_choice == "traversal";
+
+      Config::Location config = is_traversal ? Config::NETPLAY_TRAVERSAL_PORT.GetLocation() :
+                                               Config::NETPLAY_HOST_PORT.GetLocation();
+
+      m_values.emplace_back(config, ValueToString(netplay_host.value()));
+    }
+
+    if (!netplay_join.empty())
+    {
+      std::vector<std::string> join_parts = SplitString(netplay_join, ':');
+      if (join_parts.size() < 2)
+      {
+        // The user is submitting a host code
+        const std::string& host_code = join_parts[0];
+        m_values.emplace_back(Config::NETPLAY_TRAVERSAL_CHOICE.GetLocation(), "traversal");
+        m_values.emplace_back(Config::NETPLAY_HOST_CODE.GetLocation(), host_code);
+      }
+      else
+      {
+        // Ther user is submitting an IP address
+        const std::string& host = join_parts[0];
+        const std::string& port_str = join_parts[1];
+        if (!std::all_of(port_str.begin(), port_str.end(), ::isdigit) || port_str.length() > 5)
```

Comment:
This fails to handle the port being an empty string, e.g. `./dolphin-emu --netplay_join 127.0.0.1::` causes our crappy `SplitString` implementation to produce `{"127.0.0.1", ""}` and then ultimately throws an exception from the `std::stoul` below.
I think you can remove this `all_of` check and use our `TryParse` instead of `std::stoul`.